### ISSUE TITLE
04. Add coarse and site54 regression tests for metric collection

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#467]
# 04. Add coarse and site54 regression tests for metric collection

## Goal

Add targeted regression tests proving that PR5 metric collection works for the actual wrapper compositions used by fbpick coarse and site54 OOF experiments.

This issue is intentionally test-focused. It should be implemented after issues 01-03.

## Files to modify

Likely targets:

- `packages/seisai-engine/tests/test_fbpick_coarse_train_trace_memory_wiring.py`
- `tests/test_site54_coarse_experiment_runtime.py`
- `packages/seisai-engine/tests/test_train_skeleton_criterion_stats.py` if created in issue 02

### 1. Coarse train bundle exposes consumable trace-memory stats

Extend the existing coarse train wiring tests so that when `train.trace_memory_consistency.enabled=true`, the built bundle has a criterion that returns non-empty stats after two synthetic training-style calls.

Check at least:

```text
trace_memory/used_rows > 0
trace_memory/memory_stored_slot_count > 0
trace_memory/effective_weight >= 0
```

Do not require real SEG-Y files. Continue using monkeypatched tiny datasets and tiny models.

### 2. Common stats collector reaches trace-memory wrapper directly

Using the helper from issue 02:

```python
stats = consume_criterion_stats(bundle.criterion)
```

Confirm the trace-memory keys are present.

### 3. Common stats collector reaches trace-memory wrapper inside site54 wrapper

Build an object graph matching production:

```text
Site54CoarseExperimentCriterion
  -> TraceMemoryConsistencyCriterion
       -> stock criterion
```

Then call the site54 criterion twice with a synthetic batch and confirm:

```python
stats = consume_criterion_stats(site54_criterion)
```

returns trace-memory metrics.

This test should verify wrapper order: the site54 wrapper remains outermost and `TraceMemoryConsistencyCriterion` remains inside it as `base_criterion`.

### 4. Velocity-mask composition still passes masked logits into trace-memory stats path

The existing site54 tests already check that masked logits are passed to the nested base criterion. Add a stats check on top of that if practical:

- the nested trace-memory criterion should consume/update memory normally when called through site54
- stats should be collectable through the outer site54 wrapper

### 5. Disabled trace-memory config produces no trace-memory metrics

For `enabled=false`, confirm that the built criterion remains the stock criterion and `consume_criterion_stats()` returns `{}` or at least no `trace_memory/` keys.

## Constraints

- Tests must not require `segyio` unless the repository’s normal unit-test environment already installs it. Prefer monkeypatches and synthetic tensors.
- Tests must not run a full OOF or full training job.
- Keep runtime small.

## Acceptance criteria

- Relevant tests pass individually:

```bash
pytest packages/seisai-engine/tests/test_fbpick_coarse_train_trace_memory_wiring.py
pytest tests/test_site54_coarse_experiment_runtime.py
pytest packages/seisai-engine/tests/test_train_skeleton_criterion_stats.py
```

- The tests fail if the common stats collector does not traverse `.base_criterion`.
- The tests fail if `TraceMemoryConsistencyCriterion.consume_stats()` stops returning trace-memory metrics.
